js/components/common/react_modal.js.jsx

```diff
@@ -34,19 +34,86 @@ class CourseModal extends React.Component {
       course: [],
       sessions: [],
       courseTitle: "",
+      clicked: 0,
     }
   }
+  /**
+   * Change the clicked and courseId state, whenever a course link is clicked.
+   */
+  clickRelated = s => {
+    let copyState = this.state
+    copyState.courseId = s
+    copyState.clicked = 1
+    this.setState(copyState)
+  }
+
+  /**
+   * Convert the course names to <a> tags and return a list of strings and <a> tags.
+   */
+  getRelatedCourses = content => {
+    const result = []
+    if (content !== null) {
+      content.split(" ").forEach((word, i) => {
+        if (word.match(/[A-Z]{3,4}[0-9]{3}[H|Y]1/gi)) {
+          let symbol = ""
+          // check if the last character is not 1 (is a symbol). If so, remove it from word and
+          // add it as a separate string to the list.
+          const sessions = ["H1", "Y1"]
+          sessions.forEach(session => {
+            if (
+              word.indexOf(session) !== word.length - 1 &&
+              word.indexOf(session) !== -1
+            ) {
+              symbol = word.substring(word.indexOf(session) + 2)
+              word = word.substring(0, word.indexOf(session) + 2)
+            }
+          })
+
+          result.push(
+            <a
+              key={i}
+              className="course-selection"
+              onClick={() => this.clickRelated(word)}
+            >
+              {word}
+            </a>
+          )
+          result.push(symbol)
+        } else {
+          result.push(word)
+        }
+        result.push(" ")
+      })
+    }
+    return result
+  }
 
   componentDidUpdate(prevProps) {
-    if (prevProps.courseId !== this.props.courseId && this.props.courseId !== "") {
-      getCourse(this.props.courseId).then(course => {
+    let changed = 0
```

Comment:
Don't use this variable. Instead you can check whether the props changed or not when initializing `newCourse` below directly.